crates/aiken-lang/src/gen_uplc/builder.rs

```diff
@@ -1168,25 +1190,22 @@ pub fn unknown_data_to_type_otherwise(
             )
             .force()
             .lambda("__val")
-            .apply(term),
+            .apply(value),
         Some(UplcType::Bool) => Term::var("__val")
             .choose_data(
                 Term::snd_pair()
                     .apply(Term::var("__pair__"))
                     .choose_list(
-                        Term::equals_integer()
-                            .apply(Term::integer(1.into()))
+                        Term::less_than_equals_integer()
+                            .apply(Term::integer(2.into()))
                             .apply(Term::fst_pair().apply(Term::var("__pair__")))
                             .delayed_if_then_else(
-                                Term::bool(true),
-                                Term::equals_integer()
-                                    .apply(Term::integer(0.into()))
-                                    .apply(Term::fst_pair().apply(Term::var("__pair__")))
-                                    .if_then_else(
-                                        Term::bool(false).delay(),
-                                        otherwise_delayed.clone(),
-                                    )
-                                    .force(),
+                                otherwise_delayed.clone(),
+                                then.lambda(name).apply(
+                                    Term::equals_integer()
+                                        .apply(Term::fst_pair().apply(Term::var("__pair__")))
+                                        .apply(Term::integer(1.into())),
+                                ),
```

Comment:
Same here. Also, this code looks duplicated; maybe to be moved into a function?

Reply:
This is fine since it's a known type not unknown

Reply:
Yeah it's a little different and only in this place so I don't think it's worth the effort to refactor.